### PR TITLE
feat(qrcode): new plugin to generate qrcode via https://qrcode.show

### PR DESCRIPTION
**Pull request by tinogomes**
## Standards checklist:

- [x] The PR title is descriptive.
- [x] The PR doesn't replicate another PR which is already open.
- [x] I have read the contribution guide and followed all the instructions.
- [x] The code follows the code style guide detailed in the wiki.
- [x] The code is mine or it's from somewhere with an MIT-compatible license.
- [x] The code is efficient, to the best of my ability, and does not waste computer resources.
- [x] The code is stable and I have tested it myself, to the best of my abilities.

## Changes:

- adds commands `qrcode` and `qrsvg`

## Other comments:

...

### CODE REVIEW

File: plugins/qrcode/qrcode.plugin.zsh [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+qrcode () {
+  local input="$*"
+  [ -z "$input" ] && _qrcode_show_message && local input="@/dev/stdin"
+  curl -d "$input" https://qrcode.show
+}
+
+qrsvg () {
+  local input="$*"
+  [ -z "$input" ] && _qrcode_show_message && local input="@/dev/stdin"
+  curl -d "${input}" https://qrcode.show -H "Accept: image/svg+xml"
```
Comment (JuniorJPDJ) [inline plugins/qrcode/qrcode.plugin.zsh:16]:
```suggestion
  curl -d "$input" https://qrcode.show -H "Accept: image/svg+xml"
```

**Comment by robbyrussell** [comment]
@tinogomes thanks for the contribution!

**Comment by madonius** [comment]
@tinogomes I would propose switching to a local qr-code generator. What do you think?

**Comment by carlosala** [comment]
The cool thing about qrcode.show is that it does not require any installation and it has no tracking, it is OSS. I'm on the fence with this one.

**Comment by madonius** [comment]
For the tracking, it's a trust me bro guarantee. The users would have to not only trust the developer but all parties involved. 

The biggest benefit, of the local solution, however, is that the user could create a qr code without the need for an internet connection. 

About qrencode, I would have proposed it, but the issue is that  it seems to be unmaintained (?). The qrrs tool seems promising though.

**Comment by carlosala** [comment]
> For the tracking, it's a trust me bro guarantee. The users would have to not only trust the developer but all parties involved.

That's indeed true.

> The biggest benefit, of the local solution, however, is that the user could create a qr code without the need for an internet connection.

IMO, that would be a great benefit. We can check for `qrencode` (which seems to be the most common among all tools) and fallback to qrcode.show

**Comment by madonius** [comment]
@carlosala As I wrote above qrencode seems to not have gotten tlc for a while now. Last comit I see was 4 years ago.
Ofcourse it _could_ be that there is no need for anything to be fixed.
I'd be happy to supply the patch with whatever tool we decide upon.

**Pull request merged**